### PR TITLE
MDEV-29982: Improve logged error message for change buffer merge

### PR DESCRIPTION
**Pull request by tychy**
- [x] *The Jira issue number for this PR is: MDEV-29982*

## Description
Error message like 
```
[ERROR] InnoDB: The age of the last checkpoint is 991661268, which exceeds the log capacity 966365799. 
```
was confusing.  To avoid users are killing the shutdown process, I changed the message like below.

```
 [ERROR] InnoDB: The age of the last checkpoint is 991661268, which exceeds the log capacity 966365799. Shutdown is still in progress.
[Note] InnoDB: Corruption fixed by log checkpoint at LSN=xxxxx
```

## How can this PR be tested?
There are no corresponding tests.

## Basing the PR against the correct MariaDB version
- [ ] *This is a new feature and the PR is based against the latest MariaDB development branch*
- [x] *This is a bug fix and the PR is based against the earliest branch in which the bug can be reproduced*

## Backward compatibility
TODO: fill details here, if applicable, or remove the section

### CODE REVIEW

File: storage/innobase/mtr/mtr0mtr.cc [hunk 1]
```diff
@@ -1175,8 +1175,11 @@ ATTRIBUTE_COLD static void log_overwrite_warning(lsn_t age, lsn_t capacity)
     log_close_warn_time= t;
 
     sql_print_error("InnoDB: The age of the last checkpoint is " LSN_PF
-                    ", which exceeds the log capacity " LSN_PF ".",
+                    ", which exceeds the log capacity " LSN_PF "."
+                    " Shutdown is still in progress.",
```
Comment (dr-m) [inline storage/innobase/mtr/mtr0mtr.cc:1179]: This message can be output also during normal operation, not only during shutdown. We would need a check for `srv_shutdown_state != SRV_SHUTDOWN_NONE` here.

I was going to fix this myself. The code is a bit different in 10.5.

File: storage/innobase/mtr/mtr0mtr.cc [hunk 2]
```diff
@@ -1175,8 +1175,11 @@ ATTRIBUTE_COLD static void log_overwrite_warning(lsn_t age, lsn_t capacity)
...
                     age, capacity);
+    sql_print_information("InnoDB: Corruption fixed by log checkpoint at LSN=" LSN_PF ".",
+                          checkpoint);
```
Comment (dr-m) [inline storage/innobase/mtr/mtr0mtr.cc:1182]: This message should actually be output by `log_t::write_checkpoint()` and only in the case the previous warning was issued. I was going to replace the `log_close_warned` with an `Atomic_relaxed<bool>` data member in `log_t`.

**Review by dr-m — COMMENTED** [review]
Thank you for your help, but I was actually going to fix this myself. I expect some merge conflicts between 10.5, 10.6, and 10.8.

File: storage/innobase/mtr/mtr0mtr.cc (hunk 1 above)
Comment (tychy) [reply storage/innobase/mtr/mtr0mtr.cc:1179]: Now I see you are talking about this. In 10.5 https://github.com/MariaDB/server/blob/3278c57917642fa5ab9b88f7e56ad3b1e364e2c4/storage/innobase/mtr/mtr0mtr.cc#L955-L957

**Comment by dr-m** [comment]
This was fixed by e0e096faaad2aee1717a955c2b4b9100f1ed7749.

**Pull request closed without merging**